### PR TITLE
[SPARK-40500][PS] Deprecate `iteritems` in DataFrame and Seriese

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?

1. Use `pd.items` instead of `pd.iteritems`
2. Deprecate `ps.iteritems`

### Why are the changes needed?
`pd.iteritems` is deprecated in 1.5

before:
```
In [4]: import pyspark.pandas as ps

In [5]: ps.Series([3, 4, 1, 1, 5])
/Users/ruifeng.zheng/Dev/spark/python/pyspark/pandas/internal.py:1573: FutureWarning: iteritems is deprecated and will be removed in a future version. Use .items instead.
  fields = [
/Users/ruifeng.zheng/Dev/spark/python/pyspark/sql/pandas/conversion.py:486: FutureWarning: iteritems is deprecated and will be removed in a future version. Use .items instead.
  for column, series in pdf.iteritems():
                                                                                
0    3
1    4
2    1
3    1
4    5
dtype: int64
```

after:
```
In [1]: import pyspark.pandas as ps

In [2]: ps.Series([3, 4, 1, 1, 5])
                                                                                
0    3
1    4
2    1
3    1
4    5
dtype: int64
```

### Does this PR introduce _any_ user-facing change?
Eliminate `iteritems` warnings

### How was this patch tested?
existing UT

### CODE REVIEW

**Review by HyukjinKwon — APPROVED** [review]

**Review by srowen — COMMENTED** [review]
Seems OK, but there also seem to be more usages in the code, like in frame.py and test_dataframe.py - are those usages the same, also deprecated?

**Comment by zhengruifeng** [comment]
remaining `iteritems`s in `frame.py` and `test_dataframe.py` are the definition and tests of PS's `iteritems` itself, so I think we should not modify them.

as to the deprecation of PS's `iteritems`, I think we can deprecate them now, WDYT @itholic  @HyukjinKwon @Yikun

**Comment by HyukjinKwon** [comment]
Yeah let's match w/ pandas

**Review by Yikun — APPROVED** [review]
> as to the deprecation of PS's iteritems, I think we can deprecate them now

Agree. And `items` are also supported by [_minimum_pandas_version (v1.0.5)](https://github.com/apache/spark/blob/master/python/setup.py#L114), so it's a safe change.

```
>>> pd.__version__
'1.0.5'
>>> pd.Series([3, 4, 1, 1, 5]).iteritems()
<zip object at 0x7f2087983280>
>>> pd.Series([3, 4, 1, 1, 5]).items()
<zip object at 0x7f2087afcb80>
```

so LGTM

**Review by itholic — APPROVED** [review]
Then maybe should we also add `.. deprecated:: 3.4.0` into the docstring of `DataFrame.iteritems` and `Series.iteritems` ??

Otherwise, LGTM.

**Review by HyukjinKwon — APPROVED** [review]

File: python/pyspark/pandas/frame.py [hunk 1]
```diff
@@ -2054,9 +2054,16 @@ def extract_kv_from_spark_row(row: Row) -> Tuple[Name, Any]:
...
-        return self.iteritems()
+    def iteritems(self) -> Iterator[Tuple[Name, "Series"]]:
+        """
+        This is an alias of ``items``.
+
+        .. deprecated:: 3.4.0
+            iteritems is deprecated and will be removed in a future version.
+            Use .items instead.
+        """
+        warnings.warn("Deprecated in 3.4, Use DataFrame.items instead.", FutureWarning)
```
Comment (HyukjinKwon) [inline python/pyspark/pandas/frame.py:2065]:
```suggestion
        warnings.warn("Deprecated in 3.4.0, Use DataFrame.items instead.", FutureWarning)
```

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by zhengruifeng** [comment]
Thank you all!

**Pull request closed without merging**